tools/bin/make/airbyte_ci_dev_install.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Check if pyenv is installed, and install it if not
+if ! which pyenv >/dev/null; then
+    echo "pyenv not found, installing pyenv..."
+    brew install pyenv
+    echo "pyenv installed."
+else
+    echo "pyenv is already installed."
+fi
```

Comment:
I'm not sure pyenv is a requirement here... I think the requirement is that the current Python version is 3.10, and pyenv is a tool to switch Python version. `pipx install --editable --force --python=python3.10 airbyte-ci/connectors/pipelines/` will fail if the user is not on `python3.10`.

My suggestion is to make this script only check that the output of `python --version` matches `Python 3.10.*` . In case of error we can suggest users to: 
* `brew install pyenv` 
* `pyenv install 3.10.12`
* `pyenv shell 3.10.12`

But we actually don't care if users use `pyenv` or any other tool to install python 3.10.

Reply:
Moreover, if executed from the repo root, pyenv won't detect that the version we want to use is 3.10 as its defined in `airbyte-ci/.python-version`

Reply:
Good call out, also worth specificying we do have a .python-version file in the root of the repo set to 3.10
https://github.com/airbytehq/airbyte/blob/89f6ecc0d0244cb24514f6011aa0d84a9b43f05b/.python-version#L0-L1